builder/simplify.cc

```diff
@@ -371,10 +399,16 @@ class simplifier : public node_mutator {
       return;
     }
 
-    // First, prune any dead lets, and any lets that are just vars
+    // - Prune dead lets
+    // - Inline single-ref lets outside of a loop, along with lets that are trivial
     for (auto it = lets.begin(); it != lets.end();) {
```

Comment:
If I'm not missing something, I think this is now identical to the `let` visitor (again). So you could try refactoring this to be a template `visit_let(const T*)`, and just call the template function twice for `let` and `let_stmt`.

Reply:
They are not the same, one has 
```
interval_expr body_bounds;
expr body = mutate(op->body, &body_bounds);
```

another

```
stmt body = mutate(op->body);
```

Reply:
Right, never mind this.